### ISSUE TITLE
styleRefresh should defer changes if only throttled tiddlers have changed

### ISSUE THREAD

**Issue by saqimtiaz** [Jermolene/TiddlyWiki5#6914]
While we defer kicking off the main refresh process when only throttled tiddlers have changed, the `styleRefresh`  makes no such distinction and still runs the refresh check for styles on every change. 

The impact may be negligible but this is probably worth addressing anyway.

**Comment by Jermolene** [comment]
Thanks @saqimtiaz that makes sense.

One possibility is that we could approach this by refactoring the throttling into the wiki object event dispatch mechanism, avoiding issuing change events entirely in the event of only throttled tiddlers being changed. It would allow us to consolidate the change coalescing processing into one place.

**Comment by saqimtiaz** [comment]
My first instinct was to consolidate the change handling in render.js but yes, doing so in the wiki object even dispatch mechanism might be more elegant.

Another associated thought is whether we might want to move the criteria for deferring changes into a configurable filter. We would need to consider whether there would be a performance impact for evaluating a filter instead of the current check against the tiddler object:

https://github.com/Jermolene/TiddlyWiki5/blob/f0423c20b9137538150954b1d8836c058a5e56a3/core/modules/startup/render.js#L88-L93

**Comment by Jermolene** [comment]
> Another associated thought is whether we might want to move the criteria for deferring changes into a configurable filter. We would need to consider whether there would be a performance impact for evaluating a filter instead of the current check against the tiddler object:

Yes indeed. I guess it would actually need to be a cascade to make it easier for plugins to insert their own rules.

**Comment by yaisog** [comment]
As I outlined here, there are benefits to having the styleRefresh not throttled.

I do see why you would like to consolidate the refresh handling, though. All I'm asking is that you keep this usecase in mind, i.e. make separate filters for styleRefresh and mainRefresh. Initially, these could be equivalent, but this should be adjustable.

Also, don't forget that the current modals don't implement refresh throttling, either.

**Comment by yaisog** [comment]
Hi @saqimtiaz, of course and apologies.
This gist of it is the following:
In wikis where refresh is a bit sluggish due to their size or complicated filter calculations, it can be beneficial to have a much quicker styleRefresh than mainRefresh to maintain the appearance of responsiveness: A tiddler with the field `throttle.refresh` is modified in response to user interaction (e.g. `mouseenter` events or keystrokes). Quick successive modifications (a selection follows the mouse, or quickly scrolling through a list using the keyboard) do not result in a mainRefresh, but each of them in a styleRefresh. Therefore, a tiddler tagged `$:/tags/Stylesheet` that transcludes the throttled tiddler will re-evaluate in quick succession and update the CSS/UI without triggering the slower mainRefresh until the TypingTimeout expires.

A method to update UI styles while bypassing the mainRefresh would be very dear to me.

**Comment by saqimtiaz** [comment]
Thank you @yaisog. I agree that it is a compelling use case that should be kept in mind.

**Comment by pmario** [comment]
related to: 

- #8677
